saleor/csv/utils/export.py

```diff
@@ -85,24 +93,71 @@ def get_product_queryset(scope: Dict[str, Union[str, dict]]) -> "QuerySet":
     elif "filter" in scope:
         queryset = ProductFilter(data=scope["filter"], queryset=queryset).qs
 
-    queryset = queryset.order_by("pk").prefetch_related(
-        "attributes", "variants", "collections", "images", "product_type", "category"
-    )
+    queryset = queryset.order_by("pk")
 
     return queryset
 
 
-def create_csv_file_and_save_in_export_file(
-    export_data: List[Dict[str, Union[str, bool]]],
+def queryset_in_batches(queryset):
+    """Slice a queryset into batches.
+
+    Input queryset should be sorted be pk.
+    """
+    start_pk = 0
+
+    while queryset.filter(pk__gt=start_pk).exists():
```

Comment:
This executes two additional queries per batch (one for `.exists()`, one for `pks[...]`). An optimized version could be something like this:

```python
while True:
    qs = queryset.filter(pk__gt=start_pk)[:BATCH_SIZE]
    pks = list(qs.values_list("pk", flat=True))
    if not pks:
        break
    yield pks
```

Reply:
I'm testing it right now, and yes, it produces fewer queries. For 200k products, it's 100 db queries instead of 122, but time is about 35 seconds instead of 30 seconds. It's a small difference. But I'll be checking if the offset takes longer than filter or just my system is overwhelmed right now.

Reply:
I changed the solution to your suggestion.

Reply:
I hope my version doesn't use OFFSET. As far as I know both use LIMIT: the original to calculate `pks[BATCH_SIZE-1]`. my version to slice `...[:BATCH_SIZE]`.

Reply:
Yes, according to Django docs is LIMIT. My mistake 😉